### PR TITLE
refactor(core): rename query "selector" to "locator"

### PR DESCRIPTION
**Pull request by jelbourn**
The first param for Angular queries (`@ViewChild` etc.) is currently named "selector". This is confusingly overloaded with CSS selectors, especially given that we _use_ CSS selectors in other parts of the framework. The term "locator" here is both accurate and easier to explain.

### CODE REVIEW

**Review by AndrewKushnir — APPROVED** [review]

**Review by AndrewKushnir — APPROVED** [review]
Reviewed-for: public-api

**Review by pkozlowski-opensource — APPROVED** [review]
LGTM

Reviewed-for: public-api

**Review by dylhunn — APPROVED** [review]
reviewed-for: fw-core, public-api

File: goldens/public-api/core/index.md [hunk 1]
```diff
@@ -1185,9 +1185,9 @@ export interface Query {
     // (undocumented)
     isViewQuery: boolean;
     // (undocumented)
-    read: any;
+    locator: any;
```
Comment (atscott) [inline goldens/public-api/core/index.md:1188]: Isn't this one a breaking change? Should it be done in a separate PR? This isn't just a parameter name, but a property name on an object.

**Comment by jelbourn** [comment]
Abandoning this since it ended up being a breaking change.
